doc/loganalyzer_to_pytest.md

```diff
@@ -0,0 +1,108 @@
+
+**Implement**:
+Module "log_processor.py" with class "LogProcessor".
+"log_processor.py" - python module which is running on the DUT. Module perform some actions based on script input parameters. It will be based on the legacy loganalyzer.py.
+Supported actions: init, analyze, extract_syslog
+- init and analyze are described above (legacy).
+- extract_syslog - Extract all syslog entries since the latest start marker (new).
+
+"LogProcessor" class interface:
+- Includes legacy "LogAnalyzer" class from loganalyzer.py module.
+- New:
+- extract_syslog(dst="/tmp/syslog") - Extract all syslog entries since the latest start marker and store them to "/tmp/syslog" folder on the DUT.
+
+Usage example:
+
+	def test(localhost, ansible_adhoc, testbed):
+		loganalyzer = Loganalyzer(match="{PATH}/loganalyzer_common_match.txt", expect="{PATH}/loganalyzer_common_expect.txt", ignore="{PATH}/loganalyzer_common_ignore.txt")
+		loganalyzer.init()
+		# Perform test steps
+		result = loganalyzer.analyze()
+		if result["summary"]["TOTAL MATCHES"] != 0:
+			loganalyzer.fetch_fail_artifacts(dst="test/{}".format(loganalyzer.unic_test_name))
+		pytest.fail("SOME_MESSAGE\n{}\n{}".format(result["summary"], result["match_messages"]))
+
+#### Approach 2
+#### Requirements
+Move existed loganalyzer to be supported by pytest framework.
+
+#### Overview
+At this approach all the work can be done on the Ansible host if script will download extracted syslog file to the Ansible host.
+
+Current loganalyzer implementation description:
+https://github.com/Azure/SONiC/wiki/LogAnalyzer
+
+#### Development
+Module "loganalyzer.py" with class "Loganalyzer".
+
+"Loganalyzer" class interface:
+- load_common_config() - Load regexps from common configuration files: match, expect and ignore which are located in some configuration directory or with current module.
+- get_search_regexp() - returns dictionary with defined regular expressions which are used for logs processing.
+- update_match(action="pop/add", type="match/expect/ignore", *args) - add or remove regexp for syslog processing.
+Supported parameters values:
+action: "pop", "add"
+type: "match", "expect", "ignore"
+args: *args
+- init() - Add start marker to the DUT syslog.
+- analyze() - Extract syslog and copy it to ansible host. Analyze extracted syslog files localy. Return python dictionary object.
+Return example:
+		{"summary":
+		"counters": {"match": 1, "expected_match": 0, "expected_missing_match": 0},
+		"match_files": {"/tmp/syslog": {"match": 0, "expected_match": 32},
+						"/tmp/syslog1": {"match": 0, "expected_match": 15}},
+						"match_messages": {"/tmp/syslog1": ["Message 1", "Message 2", "Message n"],
+						 "/tmp/syslog2": ["Message 1", "Message 2", "Message n"]}
+	}
+- run_cmd(callable) - call function and analyze DUT logs, return the same result as "analyze" function
+
+Usage example:
+
+	from loganalyzer import Loganalyzer
+	def test(localhost, ansible_adhoc, testbed):
+		loganalyzer = Loganalyzer()

```

Comment:
What is the default in this case, why load_common_config is a different method? What if the user will not call load_common_config/update_match?

Reply:
In this approach, when loganalyzer is initializing:
loganalyzer = Loganalyzer()

It does not have configured search regexp (does not know what markers to search in the logs).
As we have common regular expressions for the DUT (now defined in match, expect and ignore files) they can be loaded/read to the loganalyzer  object by calling "loganalyzer.load_common_config()".

In most cases we will need common regexp to be loaded so this can be moved to the "__init__".

"update_match()" - not required to be used. It can be used if there is a need to remove or update (add test specific) configured regular expressions to search.